Source/WebKit/UIProcess/API/glib/WebKitImagePrivate.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2025 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#pragma once
+
+#include "WebKitImage.h"
+#include <WebKit/WKBase.h>
+#include <wtf/glib/GRefPtr.h>
+
+using WebKitImageCreateBytesFunction = WTF::Function<GRefPtr<GBytes>(WebKitImage*)>;
+
+WK_EXPORT WebKitImage* webkitImageNew(int width, int height, guint stride, const CString& type, WebKitImageCreateBytesFunction&& createBytesFn);
```

Comment:
I would use `WTF::CompletionHandler`, which is similar to `WTF::Function` but guarantees that it will be only ever invoked once&mdash;which is the case here.

Also, probably not worth it to define the `WebKitImageCreateBytesFunction` type alias, given that it is used only in a few places.


```suggestion
WK_EXPORT WebKitImage* webkitImageNew(int width, int height, guint stride, CompletionHandler<GRefPtr<GBytes>(WebKitImage*)>&& createBytes);
```

Note how the `WTF::` namespace prefix is not needed: most declarations provided by WTF are automatically placed into the global namespace for the sake of convenience, and using the prefix is rarely needed when we need to disambiguate.

Reply:
It seems that `CompletionHandler` wants to run on the main thread, but if we do this, we would defeat the purpose of the `load_async`, wouldn't we?

I will check if there's a completion handler equivalent -- which seems to be an std::exchange with null to ensure the single invokation -- that doesn't impose a thread requirement.

I see there's a second argument, `CompletionHandlerCallThread` :+1: